yarn-project/end-to-end/src/e2e_p2p/slashing.test.ts

```diff
@@ -0,0 +1,278 @@
+      t.logger.info(`We have ${sInfo.leaderVotes} votes in round ${sInfo.roundNumber} on ${sInfo.info[1]}`);
+      if (sInfo.leaderVotes > votesNeeded) {
+        t.logger.info(`We have sufficient votes`);
+        break;
+      }
+    }
+
+    t.logger.info('Deploy the actual payload for slashing!');
+    const slashEvent = slasher.slashEvents[0];
+    await t.ctx.deployL1ContractsValues.publicClient.waitForTransactionReceipt({
+      hash: await slashFactory.write.createSlashPayload([slashEvent.epoch, slashEvent.amount], {
+        account: t.ctx.deployL1ContractsValues.walletClient.account,
+      }),
+    });
+
+    t.logger.info(`We jump in time to the next round to execute`);
+    const nextRoundTimestamp2 = await rollup.read.getTimestampForSlot([
+      ((await rollup.read.getCurrentSlot()) / roundSize) * roundSize + roundSize,
+    ]);
+    await t.ctx.cheatCodes.eth.warp(Number(nextRoundTimestamp2));
+    await waitForL1Block();
+
+    const attestersPre = await rollup.read.getAttesters();
+
+    for (const attester of attestersPre) {
+      const attesterInfo = await rollup.read.getInfo([attester]);
+      // Check that status isValidating
+      expect(attesterInfo.status).toEqual(1);
+    }
+
+    t.logger.info(`Push the proposal, SLASHING!`);
+    const tx = await slashingProposer.write.pushProposal([sInfo.roundNumber], {
+      account: t.ctx.deployL1ContractsValues.walletClient.account,
+    });
+    await t.ctx.deployL1ContractsValues.publicClient.waitForTransactionReceipt({
+      hash: tx,
+    });
+
+    const receipt = await t.ctx.deployL1ContractsValues.publicClient.getTransactionReceipt({
+      hash: tx,
+    });
+
+    const slashingEvents = parseEventLogs({
+      abi: RollupAbi,
+      logs: receipt.logs,
+    }).filter(log => log.eventName === 'Slashed');
+
+    const attestersSlashed = slashingEvents.map(event => {
+      // Because TS is a little nagging bitch
+      return (event.args as any).attester;
+    });
+
+    // Convert attestersPre elements to lowercase for consistent comparison
+    const normalizedAttestersPre = attestersPre.map(addr => addr.toLowerCase());
+    const normalizedAttestersSlashed = attestersSlashed.map(addr => addr.toLowerCase());
+    expect(new Set(normalizedAttestersPre)).toEqual(new Set(normalizedAttestersSlashed));
+
+    const instanceAddress = t.ctx.deployL1ContractsValues.l1ContractAddresses.rollupAddress.toString();
+    const infoPost = await slashingProposer.read.rounds([instanceAddress, sInfo.roundNumber]);
+
+    expect(sInfo.info[0]).toEqual(infoPost[0]);
+    expect(sInfo.info[1]).toEqual(infoPost[1]);
+    expect(sInfo.info[2]).toEqual(false);
+    expect(infoPost[2]).toEqual(true);
+
+    const attestersPost = await rollup.read.getAttesters();
+
+    for (const attester of attestersPre) {
+      const attesterInfo = await rollup.read.getInfo([attester]);
+      // Check that status is Living
+      expect(attesterInfo.status).toEqual(2);
```

Comment:
cant cast an enum here?

Reply:
The num is in sol, we don't got it on the ts side.